relayer/src/keyring/pub_key.rs

```diff
@@ -63,7 +63,11 @@ impl FromStr for EncodedPubKey {
                     proto.tpe
                 );
 
-                if proto.tpe != "/cosmos.crypto.secp256k1.PubKey" {
+                // Ethermint pubkey types:
+                // "/ethermint.crypto.v1alpha1.ethsecp256k1.PubKey", "/injective.crypto.v1beta1.ethsecp256k1.PubKey"
+                if proto.tpe != "/cosmos.crypto.secp256k1.PubKey"
+                    && !proto.tpe.ends_with(".ethsecp256k1.PubKey")
```

Comment:
Related to my previous comment: would it make more sense to be more conservative in what we accept here? We could have these two `ethsecp256k1` pubkey type values defined either as enum variants or constants, and then match against them here.